pkg/platform/kube/test/platform_test.go

```diff
@@ -1698,25 +1699,57 @@ def handler(context, event):
 	suite.DeployFunctionAndRedeploy(createFunctionOptions, afterFirstDeploy, afterSecondDeploy)
 }
 
-func (suite *DeleteFunctionTestSuite) TestProcessorShutdown() {
+func (suite *DeployFunctionTestSuite) TestProcessorShutdown() {
 	functionName := "func-to-test-processor-shutdown"
 	createFunctionOptions := suite.CompileCreateFunctionOptions(functionName)
 
+	// change source code
+	createFunctionOptions.FunctionConfig.Spec.Build.FunctionSourceCode = base64.StdEncoding.EncodeToString([]byte(`
+terminated = False
+def handler(context, event):
+    if terminated:
+        context.logger.info("Got event after termination")
+    else:
+        context.logger.info("Got http event")
+    return "got http event!"
+
+def callback():
+    import time
+    time.sleep(3)
+    terminated = True
+
+def init_context(context):
+    context.platform.set_termination_callback(callback)
+`))
+
 	suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
 		suite.Require().NotEmpty(deployResult)
+
 		pods := suite.GetFunctionPods(functionName)
 		firstPod := pods[0]
 		err := suite.WaitMessageInPodLog(firstPod.Namespace, firstPod.Name, "Processor started", &v1.PodLogOptions{}, 20*time.Second)
 		suite.Require().NoError(err)
 
+		url := fmt.Sprintf("http://%s:%d", suite.GetTestHost(), deployResult.Port)
+		_, err = http.Get(url)
+		suite.Require().NoError(err)
```

Comment:
Move it to a helper function in this file `invokeFunction`, which can generally send a body, and perform several http methods.